x-pack/filebeat/input/gcppubsub/input.go

```diff
@@ -250,3 +245,20 @@ func (in *pubsubInput) getOrCreateSubscription(ctx context.Context, client *pubs
 
 	return nil, errors.New("no subscription exists and 'subscription.create' is not enabled")
 }
+
+func (in *pubsubInput) newPubsubClient(ctx context.Context) (*pubsub.Client, error) {
+	opts := []option.ClientOption{option.WithUserAgent(useragent.UserAgent("Filebeat"))}
+	if in.PubsubAlternativeHost != "" {
+		// this will be typically set because we want to point the input to a testing pubsub emulator
+		conn, err := grpc.Dial(in.PubsubAlternativeHost, grpc.WithInsecure())
+		if err != nil {
+			return nil, fmt.Errorf("grpc.Dial: %v", err)
+		}
+		opts = append(opts, option.WithGRPCConn(conn), option.WithTelemetryDisabled())
+	} else if in.CredentialsFile != "" {
```

Comment:
The PubsubAlternativeHost and Credentials* options are not mutually exclusive, right?
```suggestion
	}
	if in.CredentialsFile != "" {
```